application/common/models/User.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace common\models;
+
+use common\helpers\Utils;
+use Exception;
+use yii\behaviors\AttributeBehavior;
+use yii\db\ActiveRecord;
+use yii\helpers\ArrayHelper;
+
+class User extends UserBase
+{
+    const SCENARIO_SAVE            = 'save';
+    const SCENARIO_UPDATE_PASSWORD = 'update_password';
+    const SCENARIO_AUTHENTICATE    = 'authenticate';
+
+    public $password;
+
+    public function scenarios(): array
```

Comment:
I couldn't tell by a quick reading of [the docs for `scenarios()`](http://www.yiiframework.com/doc-2.0/yii-base-model.html#scenarios()-detail), but if a given scenario doesn't include a specific field, can it be massively assigned or not? It sounded like it would simply not be validated... which seems odd, and isn't what we want (I think). The comment in the docs about using `!fieldName` to make it non-massively-assignable is what makes me wonder...